modules/core/src/main/java/org/apache/ignite/internal/processors/platform/client/ClientRequestHandler.java

```diff
@@ -115,7 +116,14 @@
         int status = e instanceof IgniteClientException ?
             ((IgniteClientException)e).statusCode() : ClientStatus.FAILED;
 
-        return new ClientResponse(req.requestId(), status, e.getMessage());
+        boolean fullStack = false;
+
+        if (X.getCause(e) != null || !X.getSuppressedList(e).isEmpty()) {
+            fullStack =
+                ctx.kernalContext().sqlListener().showFullStackOnClientSide();
+        }
+
+        return new ClientResponse(req.requestId(), status, fullStack ? X.getFullStackTrace(e) : e.getMessage());
```

Comment:
I think the first line should always be the same - `e.getMessage()`. When stack traces are enabled, append stack trace after a newline separator.